src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<K, V> {
+    private K first;
+    private V second;
+
+    private Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <K, V> Pair<K, V> of(K first, V second) {
+        return new Pair<>(first, second);
+    }
+
+    public K getFirst() {
+        return first;
+    }
+
+    public V getSecond() {
+        return second;
+    }
+
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Pair<K, V> otherPair = (Pair<K, V>) object;
+        return first == null ? otherPair.first == null : first.equals(otherPair.first)
```

Comment:
try to make this check without ternary operator, just use logical operators && and ||

Reply:
done!)